doc-site/docs/architecture/distributed_sequencer.md

```diff
@@ -0,0 +1,1437 @@
+
+![image](../images/distributed_sequencer_partition_frame2.svg){ width="500" }
+
+At this point, node B and node C independently conclude that node A is the preferred coordinator and accordingly, they delegate their transactions to node A.  Node C ceases to act as coordinator and abandons any transactions that it has in flight.
+
+![image](../images/distributed_sequencer_partition_frame3.svg){ width="500" }
+
+The system reaches the desired eventual state of node A being coordinator, sending heartbeat messages to all nodes
+
+![image](../images/distributed_sequencer_partition_frame4.svg){ width="500" }
+
+TODO
+
+ - might be useful to illustrate the "limp mode" in more detail where both coordinators are managing to get transactions through but the base ledger contract is providing double spend protection and the retry strategy of the coordinator means that transactions are eventually processed correctly 
+ - all of the above seems to assume that the network partition affects communication between paladin nodes but not the blockchain.  Should really discuss and elaborate what happens when the block chain nodes are not able to communicate with each other ( esp. in case of public chains where long finality times are possible).  The TL;DR is - notwithstanding this algorithm, the paladin core engine and its management of domain contexts must be mindful of the difference between block receipts and finality and should have mechanisms to configure the number of confirmed blocks it considers as final and/or to reset domain contexts and redo transaction intents that ended up on blocks that got reversed.
+ 
+### Coordinator switchover
+When the block height reaches a range boundary then all nodes independently reach a universal conclusion the new coordinator.
+
+Lets consider the case where the scenario explored above continues to the point where the blockchain moves to block 4.  In the interim, while node A was coordinator, nodes B, C and D delegated transactions `B-1`, `B-2`, `B-3`,  `C-1`, `C-2`, `C-3`, `D-1`, `D-2` and `D-3`,  and A itself has assembled `A-1`, `A-2`, `A-3`.
+
+`A-1` , `B-1`, `C-1` and `D-1` have already been submitted to base ledger and confirmed.  
+`A-2` , `B-2`, `C-2` and `D-2` have been prepared for submission, assigned a signing address and a nonce but have not yet been mined into a confirmed block.  NOTE it is irrelevant whether these transactions have been the subject of an `eth_sendTransaction` yet or not.  i.e. whether they only exist in the memory of the paladin node's transaction manager or whether they exist in the mempool of node A's blockchain node.  What is important is that
+
+ - they have been persisted to node A's database as being "dispatched" and
+ - are not included in block 3 or earlier
+
+![image](../images/distributed_sequencer_switchover_frame1.svg){ width="500" }
+
+As all nodes in the group ( including node A) receive receipts of block 3, they will recognize node B as the new coordinator
```

Comment:
I did not understand this sentence.

This implies some future inference model and/or state tracking that seems very complex to achieve, and it's unclear to me the justification for this.

I note I believe this different statement is true, but it's currently the only thing I understand to be true with respect to evaluation of block height:

> When each nodes in the group is notified that the block height of the chain has reached 4 (happens at different times on each node) they will recognize node B as the highest scoring coordinator when they have reason to run the scoring algorithm

Reply:
Ah, ok - I think I understand.
You are saying that the the combination of these two things will cause the scoring algorithm to occur:
- Detection of block height moving to 4
- The fact that transactions are unresolved for this contract address

This implies the algorithm re-evaluates the local "sender state" on detection of each block.

However, I'm worried I'm missing some detail as you say:

> receive **receipts** of block **3**

I'm not clear what receiving receipts means in this case, how it relates to block confirmations, or why it's block 3 (rather than 4) that's relevant here.

Reply:
The block numbers in the diagram start at 0 therefore block 3 is the last block in the first range.  Early versions of these diagrams had `x+0`,`x+1` etc... where `x` was the block height when the privacy group was created but I felt that was adding noise to the diagrams with no added value so I simplified it to be an absolute numbering and it happened to start at 0.

So when we see that block 3 has been mined, we can assume that any further transactions should come into the second range.  Given that block 0 is the genesis block, it is probably not very helpful for me to be including it in the first range so I'll change it the numbering system to start at 1.

Reply:
The indexing wasn't the problem. Starting from zero was fine for me - we were referring to the same block.

I think you just spelt "Detection of block height moving to 4" as "receive receipts of block 3".

If instead you had said "confirmation of block 3 being mined, so the next block is now block 4" I'd have been all square.